src/controllers/krate/publish.rs

```diff
@@ -633,6 +608,36 @@ async fn read_json_metadata(
         .map_err(|e| bad_request(format_args!("invalid upload request: {e}")))
 }
 
+async fn read_tarball_bytes(
+    reader: &mut (impl AsyncReadExt + Unpin),
+    max_length: u32,
+) -> Result<Bytes, BoxedAppError> {
```

Comment:
I think `use tokio::io::AsyncReadExt` with the following should be enough?
```suggestion
async fn read_tarball_bytes<R: AsyncRead + Unpin>(
    reader: &mut R,
    max_length: u32,
) -> Result<Bytes, BoxedAppError> {
```